02-spatial-data.Rmd

```diff
@@ -444,6 +444,74 @@ plot(st_as_sfc("GEOMETRYCOLLECTION (MULTIPOINT (5 2, 1 3, 3 4, 3 2), LINESTRING
 par(old_par)
 ```
 
+### The sf class {#sf}
+
+Simple features consist of two main parts: geometries and non-geographic attributes.
+Figure \@ref(fig:02-sfdiagram) shows how an sf object is created -- geometries come from an `sfc` object, while attributes are taken from a `data.frame` or `tibble`.
+To learn more about building sf geometries from scratch read the following Sections \@ref(sfg) and \@ref(sfc).
+
+```{r 02-sfdiagram, fig.cap="Building blocks on sf objects."}
+knitr::include_graphics("figures/02-sfdiagram.png")
```

Comment:
Only comment here: shouldn't we add 

```r
# source("code/02-sfdiagram.R")
```

Above the line to show the code that generates the diagram?